File: api/Models/Review.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace api.Models
{
    public class Review
    {
        public int Id { get; set; }

        public string UserId { get; set; }
        public ApplicationUser User { get; set; } = null!;

        public int ServiceId { get; set; }
        public Service Service { get; set; } = null!;

        [Range(1, 5)]
        public int Rating { get; set; }

        public string? Comment { get; set; }

        [Column(TypeName = "datetime")]
        public DateTime DateTime { get; set; }
    }
}

File: src/pages/users/review/review.js
import { memo } from 'react';
import React, { useState, useEffect } from 'react';
import { getAllReviews } from '../../../api/review';

const Review = () => {
  const [reviews, setReviews] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState('');

  useEffect(() => {
    const fetchReviews = async () => {
      try {
        const data = await getAllReviews();
        setReviews(data);
      } catch (error) {
        setError(`Error fetching reviews: ${error.message}`);
      } finally {
        setLoading(false);
      }
    };
    fetchReviews();
  }, []);

  if (loading) {
    return <p>Loading...</p>;
  }

  if (error) {
    return <p>{error}</p>;
  }
  return (
    <div>
      <h2>Reviews</h2>
      <ul>
        {reviews.map((review) => (
          <li key={review.id}>
            ID: {review.id}<br />
            UserID: {review.userId}<br />
            serviceID: {review.serviceId}<br />
            Rating: {review.rating}<br />
            Comment: {review.comment}<br />
            Date: {review.dateTime}<br />
          </li>
        ))}
      </ul>
    </div>
  )
}

export default memo(Review);